coding_systems/icd10/import_data.py

```diff
@@ -17,8 +19,67 @@ def import_data(release_path):
 
 
 def load_concepts(doc):
+    """Yield dicts with `code`, `kind`, `term` and `parent_id` attributes of all
+    concepts.
+
+    Concepts are defined in `Class` elements, which have `code` and `kind` attributes.
+    They have one or more `Rubric` elements as children, and we choose one of them for
+    the term.  Additionally, non-chapter `Class` elements have a single `SuperClass`
+    child, which contains the `parent_id`.
+
+    Several `Class` elements also have a `ModifiedBy` child, which references a number
+    of `ModifierClass` elements.  For instance, we have:
+
+    <ClaML version="2.0.0">
+        <Class code="E13" kind="category">
+            <SuperClass code="E10-E14"/>
+            <ModifiedBy code="S04E10_4"/>
+            <Rubric kind="preferred">
+                <Label>Other specified diabetes mellitus</Label>
+            </Rubric>
+        </Class>
+        ...
+        <ModifierClass code=".0" modifier="S04E10_4">
+            <SuperClass code="S04E10_4"/>
+            <Rubric kind="preferred">
+                    <Label>With coma</Label>
+            </Rubric>
+        </ModifierClass>
+        <ModifierClass code=".1" modifier="S04E10_4">
+            <SuperClass code="S04E10_4"/>
+            <Rubric kind="preferred">
+                <Label>With ketoacidosis</Label>
+            </Rubric>
+        </ModifierClass>
+        ...
+    </ClaML>
+
+    This means that category E13 has descendants E13.0 and E13.1, which we label as
+    "Other specified diabetes mellitus : With coma" and "Other specified diabetes
+    mellitus : With ketoacidosis".
+
+    Note that in order to match the format the ICD-10 codes are recorded in SUS and
+    elsewhere, we strip the `.` from codes.  So eg `E13.0` is recorded as `E130`.  We
+    don't think that this ever introduces ambiguity!
+    """
+
     root = doc.getroot()
 
+    modifiers = defaultdict(list)
+
+    for e in root.findall("ModifierClass"):
+        modifier = e.get("modifier")
+        code = e.get("code")
+        if code[0] != ".":
```

Comment:
Would `str.startswith` be more approrpriate here, to avoid problems with `code` being an empty string?  Or do we know that `code` will always be 1+ long?

Reply:
Good suggestion, but (a) `code` is never empty and (b) we're not running 3.9 yet.

Reply:
`startswith` was added at some point between 1.6 and 2.0.

Reply:
You're very right.  I was thinking of `str.removeprefix`, or rather, I was not thinking.